### PR TITLE
[CIRC-1909] Logging - circulation domain (D - L)

### PR DESCRIPTION
**Pull request by roman-barannyk**
## Purpose
Log org.folio.circulation.domain from Department to Location in alphabetic order

Resolves: [CIRC-1909](https://issues.folio.org/browse/CIRC-1909)

### CODE REVIEW

File: src/main/java/org/folio/circulation/domain/Loan.java [hunk 1]
```diff
@@ -226,11 +241,13 @@ public String getActionComment() {
 
   public Result<Void> isValidStatus() {
     if (!representation.containsKey(STATUS)) {
+      log.warn("isValidStatus:: Loan does not have a status");
       return failedDueToServerError("Loan does not have a status");
     }
 
     // Provided status name is not present in the enum
     if (getStatus() == null) {
+      log.warn("isValidStatus:: Loan status must be \"Open\" or \"Closed\"");
```
Comment (OleksandrVidinieiev) [inline src/main/java/org/folio/circulation/domain/Loan.java:250]: Can we extract message into a variable and reuse it?

File: src/main/java/org/folio/circulation/domain/Loan.java [hunk 2]
```diff
@@ -226,11 +241,13 @@ public String getActionComment() {
 
   public Result<Void> isValidStatus() {
     if (!representation.containsKey(STATUS)) {
+      log.warn("isValidStatus:: Loan does not have a status");
```
Comment (OleksandrVidinieiev) [inline src/main/java/org/folio/circulation/domain/Loan.java:244]: Can we extract message into a variable and reuse it?

File: src/main/java/org/folio/circulation/domain/Loan.java [hunk 3]
```diff
@@ -240,6 +257,7 @@ public Result<Void> isValidStatus() {
 
   public Result<Void> openLoanHasUserId() {
     if (isOpen() && getUserId() == null) {
+      log.warn("openLoanHasUserId:: Open loan must have a user ID");
```
Comment (OleksandrVidinieiev) [inline src/main/java/org/folio/circulation/domain/Loan.java:260]: Can we extract message into a variable and reuse it?

File: src/main/java/org/folio/circulation/domain/Loan.java [hunk 4]
```diff
@@ -249,6 +267,7 @@ public Result<Void> openLoanHasUserId() {
 
   public Result<Void> closedLoanHasCheckInServicePointId() {
     if (isClosed() && getCheckInServicePointId() == null) {
+      log.warn("closedLoanHasCheckInServicePointId:: a closed loan must have a checkin service point");
```
Comment (OleksandrVidinieiev) [inline src/main/java/org/folio/circulation/domain/Loan.java:270]: Can we extract message into a variable and reuse it?

File: src/main/java/org/folio/circulation/domain/Loan.java [hunk 5]
```diff
@@ -454,6 +473,8 @@ public String getPatronGroupIdAtCheckout() {
   }
 
   public Loan renew(ZonedDateTime dueDate, String basedUponLoanPolicyId) {
+    log.debug("renew:: parameters dueDate: {}, basedUponLoanPolicyId: {}",
```
Comment (OleksandrVidinieiev) [inline src/main/java/org/folio/circulation/domain/Loan.java:476]: `ZonedDateTime#toString` is not that trivial. I would log it via Supplier.

File: src/main/java/org/folio/circulation/domain/LoanCheckInService.java [hunk 6]
```diff
@@ -6,18 +6,25 @@
...
 import org.folio.circulation.support.ServerErrorFailure;
 import org.folio.circulation.support.results.Result;
 
 public class LoanCheckInService {
+  private static final Logger log = LogManager.getLogger(MethodHandles.lookup().lookupClass());
 
   public Result<Loan> checkIn(Loan loan, ZonedDateTime systemDateTime,
     CheckInByBarcodeRequest request) {
 
+    log.debug("checkIn:: parameters loan: {}, systemDateTime: {}, request: {}",
```
Comment (OleksandrVidinieiev) [inline src/main/java/org/folio/circulation/domain/LoanCheckInService.java:24]: Use suppliers

File: src/main/java/org/folio/circulation/domain/Loan.java (hunk 1 above)
Comment (roman-barannyk) [reply src/main/java/org/folio/circulation/domain/Loan.java]: done

File: src/main/java/org/folio/circulation/domain/Loan.java (hunk 2 above)
Comment (roman-barannyk) [reply src/main/java/org/folio/circulation/domain/Loan.java]: done

File: src/main/java/org/folio/circulation/domain/Loan.java (hunk 3 above)
Comment (roman-barannyk) [reply src/main/java/org/folio/circulation/domain/Loan.java]: done

File: src/main/java/org/folio/circulation/domain/Loan.java (hunk 4 above)
Comment (roman-barannyk) [reply src/main/java/org/folio/circulation/domain/Loan.java]: done

File: src/main/java/org/folio/circulation/domain/Loan.java (hunk 5 above)
Comment (roman-barannyk) [reply src/main/java/org/folio/circulation/domain/Loan.java:477]: done

File: src/main/java/org/folio/circulation/domain/LoanCheckInService.java (hunk 6 above)
Comment (roman-barannyk) [reply src/main/java/org/folio/circulation/domain/LoanCheckInService.java:24]: done

File: src/main/java/org/folio/circulation/domain/LoanRepresentation.java [hunk 7]
```diff
@@ -23,32 +23,39 @@ public class LoanRepresentation {
...
       addAdditionalServicePointProperties(extendedRepresentation, loan.getCheckoutServicePoint(), "checkoutServicePoint");
     }
 
     if (loan.getUser() != null) {
+      log.info("extendedLoan:: user is not null");
       additionalBorrowerProperties(extendedRepresentation, loan.getUser());
-    }else{
+    } else {
       //When there is no user, it means that the loan has been anonymized
+      log.info("extendedLoan:: the is no user, removing borrower");
```
Comment (OleksandrVidinieiev) [inline src/main/java/org/folio/circulation/domain/LoanRepresentation.java:58]:
```suggestion
      log.info("extendedLoan:: there is no user, removing borrower");
```

File: src/main/java/org/folio/circulation/domain/LoanService.java [hunk 8]
```diff
@@ -23,21 +27,27 @@ public LoanService(Clients clients) {
 
   public CompletableFuture<Result<LoanAndRelatedRecords>> truncateLoanWhenItemRecalled(
     LoanAndRelatedRecords records) {
+
+    log.debug("truncateLoanWhenItemRecalled:: parameters records: {}",
+      () -> records);
```
Comment (OleksandrVidinieiev) [inline src/main/java/org/folio/circulation/domain/LoanService.java:32]:
```suggestion
    log.debug("truncateLoanWhenItemRecalled:: parameters records: {}", () -> records);
```

File: src/main/java/org/folio/circulation/domain/LoanRepresentation.java (hunk 7 above)
Comment (roman-barannyk) [reply src/main/java/org/folio/circulation/domain/LoanRepresentation.java]: corrected

File: src/main/java/org/folio/circulation/domain/LoanService.java (hunk 8 above)
Comment (roman-barannyk) [reply src/main/java/org/folio/circulation/domain/LoanService.java]: updated

**Review by OleksandrVidinieiev — APPROVED** [review]

**Review by alexanderkurash — APPROVED** [review]

**Pull request merged**